### DISCUSSION TITLE
v5 Scripts

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#881] [Announcements]
Working on creating the next version of the scripts
- **The `***-v5.sh` script serves both to create the LXC (Proxmox Shell) and provide support for the App (LXC Console)**
- FULL Verbose Mode (see every action) This should help troubleshooting.
- Reduce the amount of support scripts.
- Code refactoring.

***Share your thoughts!***

**Comment by tteck** [comment]
The v5 `***-install.sh` script isn't backwards compatible. So, a new `***-v5-install.sh` will be created so that v4 will still work. Maybe it's time to purge the repo of old versions?

**Reply by tteck** [reply]
I vote purge

**Reply by proddy** [reply]
do it!

**Reply by AlexGeek77** [reply]
Purge to keep it clean and easy for all to follow.

**Reply by Beef-Tech** [reply]
Yeap, I agree with you.

I'm working on a fork from your repo in order to use it on my Pimox install (on a Raspberry Pi), and I was thinking about suggesting that !

Also thank you for your work !

**Comment by Beef-Tech** [comment]
I was also thinking about common-ize some things :

- Like I said before, on my fork, I saw that there are a lot of duplicates between scripts (for testing Network access/config for example) and I think it could be usefull to have a "common" script to do those things.
- I would suggest also to add a check to ensure that commands that will be executed are available on the LXC template (for example, I had some errors about some Debian templates which didn't contain "nslookup" or "wget" or again "netcat" commands

**Reply by tteck** [reply]
The "Debian templates" (arm64) that you're using differ from a regular Proxmox Debian template.

**Comment by proddy** [comment]
With v5 will there still be a way to upgrade Home Assistant Core? There was previously the `misc/core-update.sh` script.

**Reply by tteck** [reply]
Of course, just follow the new update instructions

**Reply by tteck** [reply]
This is where "_Reduce the amount of support scripts._" comes in play.
I'd love to hear your opinion. As I'm still _up in the air_ with it.

**Reply by proddy** [reply]
Yeah I think it's super awesome having a single script for both install and update. I was a bit confused at first in the description 
```
To create a new Proxmox Home Assistant Core LXC, run the following in the Proxmox Shell.
To Initialize (first start), Update or Install HACS, run the following in the LXC Console.
```

but now I understand and its works perfectly.

**Reply by tteck** [reply]
Suggestions on how to make it less confusing?

**Reply by proddy** [reply]
To create a new Proxmox Home Assistant Core LXC run the command below in the Proxmox Shell.

To Initialize (first start), Update or Install the HACS Integration run the command below from the LXC Console.

?

**Reply by tteck** [reply]
Let's try it

**Reply by LubomirH** [reply]
I have just realized there were some changes in the scripts :)
I understood I need to use the consolidated https://github.com/tteck/Proxmox/raw/main/ct/homeassistant-core.sh for both creating the LXC on PVE and also for installing/updating the HAcore inside the LXC.

I am just wondering what is then the https://github.com/tteck/Proxmox/raw/main/install/homeassistant-core-install.sh for.

thanks for answer

Lubomir

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/blob/main/CODE-AUDIT.md

**Comment by timdonovanuk** [comment]
I have a feature request for the v5 scripts: 99% of these apps have a web end point so it would be nice if that endpoint was placed in the LXC notes (as a clickable URL). Something like:

![image](https://user-images.githubusercontent.com/8156439/210358892-40b4b75f-9fb1-483a-a554-d2ef588ea483.png)


Thanks :)

**Reply by tteck** [reply]
That's easily done, but if your IP changes, the IP in notes doesn't follow

**Reply by timdonovanuk** [reply]
True but if people aren't using static IP's for their services they'd never know what the URL is anyway ;) Could output only if people use a static then I guess. Cheers and happy new year!

**Reply by tteck** [reply]
Gotta think about DHCP and mDNS users.
I set the static IP (reservations) to the CTID. 
Examples: 
for CTID 100 I use IP `192.168.86.100`
for CTID 101 I use IP `192.168.86.101`
and so on
